[Python][Flaky Test] test_sync_pubsub_combined_exact_and_pattern_one_client

### Test Name

test_sync_pubsub_combined_exact_and_pattern_one_client[SubscriptionMethod.Blocking-MessageReadMethod.Callback-True]

### Test Location

tests/sync_tests/test_sync_pubsub.py::TestSyncPubSub::test_sync_pubsub_combined_exact_and_pattern_one_client

### Failure Permlink

https://github.com/valkey-io/valkey-glide/actions/runs/26735171689/job/78786853761#logs

### Frequency

Observed in scheduled CI run on 2026-06-01. Failed in 1 out of 2 Python PubSubTests jobs.

### Steps to Reproduce

1. Run the FMT Python workflow on schedule
2. Observe failure in Python PubSubTests - 3.9, EngineVersion - 6.2, Target - aarch64-unknown-linux-gnu

### System Information

GitHub Actions (aarch64-unknown-linux-gnu)

### Language and Version

Python 3.9

### Engine Version

redis 6.2

### Logs

```
tests/sync_tests/test_sync_pubsub.py::TestSyncPubSub::test_sync_pubsub_combined_exact_and_pattern_one_client[SubscriptionMethod.Blocking-MessageReadMethod.Callback-True] FAILED [ 82%]

            assert result == 1
            assert channel_str in all_channels_and_messages.keys()
            assert pubsub_msg.message == all_channels_and_messages[channel_str]
            assert pubsub_msg.pattern == pattern
            assert all_channels_and_messages == {}
            # assert there are no messages to read
            assert client.try_get_pubsub_message() is None
            assert callback is not None
>           assert len(callback) == expected_callback_messages_count
E           AssertionError

tests/utils/pubsub_test_utils.py:1137: AssertionError
```

### Screenshots

_No response_

### Glide Version

main

### Expected Behavior

Test should pass reliably.

### Actual Behavior

The callback message count does not match the expected count. The test subscribes to both exact channels and pattern channels with a single client, publishes messages, and then verifies all messages were received via callback. The assertion fails because not all expected messages arrived in the callback within the test window.

### Possible Fixes

- Add a retry/wait mechanism for callback messages to arrive before asserting count
- Increase the polling timeout for callback delivery
- This may be a race condition on aarch64 where message delivery via callback is slower

--- a/python/tests/sync_tests/test_sync_pubsub.py
+++ b/python/tests/sync_tests/test_sync_pubsub.py
@@ -812,17 +812,24 @@ class TestSyncPubSub:
                 if cluster_mode:
                     assert result == 1
 
-            # Wait for messages to propagate
+            # Wait for messages to propagate. Both the Callback and Sync read
+            # paths are non-blocking, so they must poll until every message has
+            # arrived; a fixed sleep races against slow CI (e.g. aarch64). Only
+            # the Async path (get_pubsub_message) blocks per-message and needs
+            # no pre-wait.
             if method == MethodTesting.Callback:
-                # Callback messages arrive asynchronously; poll until all are received
+                # Callback messages arrive asynchronously; poll the callback list.
                 wait_for_messages(
                     len(all_channels_and_messages), callback_messages, timeout=10.0
                 )
-            else:
-                # For Async (blocking read) and Sync (non-blocking read),
-                # allow a brief propagation window for messages to reach the
-                # internal queue before non-blocking try_get_pubsub_message calls
-                time.sleep(1)
+            elif method == MethodTesting.Sync:
+                # try_get_pubsub_message is non-blocking; poll the client's
+                # internal pubsub queue (a List[PubSubMsg]) until all arrive.
+                wait_for_messages(
+                    len(all_channels_and_messages),
+                    listening_client._pubsub_queue,
+                    timeout=10.0,
+                )
 
             # Check if all messages are received correctly
             for index in range(len(all_channels_and_messages)):

--- a/python/tests/sync_tests/test_sync_pubsub.py
+++ b/python/tests/sync_tests/test_sync_pubsub.py
@@ -812,8 +812,17 @@ class TestSyncPubSub:
                 if cluster_mode:
                     assert result == 1
 
-            # allow the message to propagate
-            time.sleep(1)
+            # Wait for messages to propagate
+            if method == MethodTesting.Callback:
+                # Callback messages arrive asynchronously; poll until all are received
+                wait_for_messages(
+                    len(all_channels_and_messages), callback_messages, timeout=10.0
+                )
+            else:
+                # For Async (blocking read) and Sync (non-blocking read),
+                # allow a brief propagation window for messages to reach the
+                # internal queue before non-blocking try_get_pubsub_message calls
+                time.sleep(1)
 
             # Check if all messages are received correctly
             for index in range(len(all_channels_and_messages)):

